apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts

```diff
@@ -7276,4 +7280,29 @@ export class ProformaInvoiceComponent implements OnInit, OnDestroy, AfterViewIni
             }
         }
     }
+
+    /**
+     * This will set the data in selected item if user is editing voucher from outside module
+     *
+     * @private
+     * @returns {Promise<void>}
+     * @memberof ProformaInvoiceComponent
+     */
+    private setSelectedItem(): Promise<void> {
+        return new Promise((resolve: Function) => {
+            this.route.queryParams.pipe(takeUntil(this.destroyed$)).subscribe((params) => {
```

Comment:
It should be subscribed only once with `take(1)` since `setSelectedItem` method will be called on every route param change which would create multiple subscriptions
Also we don't need to convert to promise here as the router events are synchronous